integration-test/tests/validator.rs

```diff
@@ -0,0 +1,169 @@
+use cosmwasm_std::Binary;
+use ethers::types::Address;
+use ethers::utils::hex::FromHex;
+use hpl_interface::{ism::multisig::ValidatorSet, types::metadata::MessageIdMultisigIsmMetadata};
+use k256::{
+    ecdsa::{RecoveryId, SigningKey, VerifyingKey},
+    elliptic_curve::rand_core::OsRng,
+};
+use ripemd::Ripemd160;
+use sha2::{Digest, Sha256};
+
+pub fn sha256_digest(bz: impl AsRef<[u8]>) -> [u8; 32] {
+    let mut hasher = Sha256::new();
+
+    hasher.update(bz);
+
+    hasher.finalize().as_slice().try_into().unwrap()
+}
+
+pub fn ripemd160_digest(bz: impl AsRef<[u8]>) -> [u8; 20] {
+    let mut hasher = Ripemd160::new();
+
+    hasher.update(bz);
+
+    hasher.finalize().as_slice().try_into().unwrap()
+}
+
+fn pub_to_addr(pub_key: Binary, prefix: &str) -> String {
+    let sha_hash = sha256_digest(pub_key);
+    let rip_hash = ripemd160_digest(sha_hash);
+
+    let addr = hpl_interface::types::bech32_encode(prefix, &rip_hash).unwrap();
+
+    addr.to_string()
+}
```

Comment:
ask; why don't we move those veirifcation functions to general utils?

Reply:
This uses hash libraries that may not be used by some contracts